File: app/routers/health.py
from fastapi import APIRouter

from app.schemas.health import HealthResponse
from app.schemas.operations import (
    CostControlStatus,
    ExternalServiceStatus,
    HealthDetailsResponse,
    StorageStatusItem,
)
from app.services.operations_service import operations_service


router = APIRouter(tags=["health"])


@router.get("/health", response_model=HealthResponse)
def health_check() -> HealthResponse:
    return HealthResponse(status="ok")


@router.get("/health/details", response_model=HealthDetailsResponse)
def health_details() -> HealthDetailsResponse:
    result = operations_service.get_health_details()
    return HealthDetailsResponse(
        status=result["status"],
        python_version=result["python_version"],
        database_url=result["database_url"],
        database_reachable=result["database_reachable"],
        ffmpeg_available=result["ffmpeg_available"],
        storage=[StorageStatusItem(**item) for item in result["storage"]],
        external_services=[ExternalServiceStatus(**item) for item in result["external_services"]],
        cost_control=CostControlStatus(**result["cost_control"]),
    )

File: frontend/src/types/health.ts
export interface HealthResponse {
  status: string;
}

export interface StorageStatusItem {
  path: string;
  exists: boolean;
  is_dir: boolean;
  writable: boolean;
}

export interface ExternalServiceStatus {
  name: string;
  configured: boolean;
  detail: string;
}

export interface CostControlStatus {
  usage_routes_available: boolean;
  ai_usage_logging_ready: boolean;
  pricing_configured: boolean;
  top_risk: string;
}

export interface HealthDetailsResponse {
  status: string;
  python_version: string;
  database_url: string;
  database_reachable: boolean;
  ffmpeg_available: boolean;
  storage: StorageStatusItem[];
  external_services: ExternalServiceStatus[];
  cost_control: CostControlStatus;
}
